sdk/python/tests/test_kfp_samples.sh

```diff
@@ -134,6 +134,12 @@ if ! (pip show "kfp-tekton" | grep Location | grep -q "${PROJECT_DIR}"); then
   pip install -q -e "${PROJECT_DIR}/sdk/python"
 fi
 
+# install pytest, unless already installed
```

Comment:
why do we need to install `pytest` here? this script is not running Python `unittest`s

Reply:
The compose and non pipeline decorator examples need to have pytests because they cannot be compiled with `dsl-compile-tekton`.

Reply:
Oh, right, Andrew had added that. Thanks. Makes sense to me now.

https://github.com/kubeflow/kfp-tekton/blob/15918434c4b40773bd14de5e71026dcc2fa64ccc/sdk/python/tests/test_kfp_samples.sh#L168-L176